homeassistant/components/tplink/switch.py

```diff
@@ -122,8 +123,23 @@ def update(self):
                 self._mac = self.smartplug.mac
                 self._alias = self.smartplug.alias
                 self._model = self.smartplug.model
-
-            self._state = self.smartplug.state == self.smartplug.SWITCH_STATE_ON
+                if self.smartplug.context is None:
+                    self._device_id = self.smartplug.sys_info["deviceId"]
+                else:
+                    self._device_id = self.smartplug.context
```

Comment:
Hmm, I'm not sure what is going to happen for those who have been using this integration when the id suddenly changes, but I wouldn't mind to keep using context here and fix the issue upstream. But in case the device id changes, is there a need to migrate them? Can someone from @home-assistant/core clarify what's the best approach?

Reply:
This doesn't need to be a breaking change.  Without knowing about this change, I re-did a lot of this work (see doc change at https://github.com/home-assistant/home-assistant.io/pull/10382 and main PR at https://github.com/home-assistant/home-assistant/pull/26655 ).

To avoid breaking change, for SmartStrip Plug (and only for those plugs), give them a unique_id mac_address + "_" + devid (or append the plug number).  Non-smart-strip plugs can just use mac like before.

Note that I didn't notice that smartstrip builds a SmartPlug for each outlet (I was reverse engineering the sys_info and the network protocol and didn't even realize that the KP400 looks like a SmartStrip because I didn't know they made those).

Reply:
Another FWIW: if you update this block (and remove the previous `self._alias` declaration) the individual outlets will use the names/aliases given to them in the Kasa app.

```python
                if self.smartplug.context is None:
                    self._device_id = self.smartplug.sys_info["deviceId"]
                    self._alias = self.smartplug.alias
                else:
                    self._device_id = self.smartplug.context
                    self._alias = [
                        child
                        for child in self.smartplug.sys_info["children"]
                        if child["id"] == self.smartplug.context
                    ][0]["alias"]
```

Reply:
Thanks again @shbatm, I've incorporated this with an additional change of setting `_device_id` to `_mac` for single plugs to retain backwards compatibility, so this will no longer will be a breaking change 👍